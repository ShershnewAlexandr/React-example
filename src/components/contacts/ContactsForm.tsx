import React from 'react';
import { Grid } from '@material-ui/core';
import { Field } from 'redux-form/immutable';
import { useTranslation } from 'react-i18next';
import TextInput from '../inputs/TextInput';
import DateInput from '../inputs/DateInput';
import PhotoInput from '../inputs/PhotoInput';
import { ContactsFormContainerProps } from '../../containers/contacts/ContactsFormContainer';

function ContactsForm({
  onloadAvatarAction,
  prerenderAvatarURL,
  handleSubmit,
  isLoading,
}: ContactsFormContainerProps) {
  const { t } = useTranslation();
  return (
    <form onSubmit={handleSubmit}>
      <Grid container direction="row" spacing={4}>
        <Grid item xs={6}>
          <Field
            name="firstName"
            component={TextInput}
            label={t('contacts.firstname')}
            disabled={isLoading}
          />
        </Grid>
        <Grid item xs={6}>
          <Field
            name="lastName"
            component={TextInput}
            type="date"
            label={t('contacts.lastname')}
            disabled={isLoading}
          />
        </Grid>
        <Grid item xs={12}>
          <Field
            name="dateOfBirthday"
            component={DateInput}
            label={t('contacts.dateOfBirthday')}
            disabled={isLoading}
          />
        </Grid>
        <Grid item xs={12}>
          <Field
            name="photo"
            component={PhotoInput}
            label={t('contacts.photo')}
            onClickAction={onloadAvatarAction}
            prerenderPhotoURL={prerenderAvatarURL}
            disabled={isLoading}
          />
        </Grid>
      </Grid>
    </form>
  );
}

export default ContactsForm;
