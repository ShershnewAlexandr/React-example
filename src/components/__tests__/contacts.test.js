import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../setupTests';
import ContactsForm from '../contacts/ContactsForm';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: str => str,
  }),
}));

describe('contactsForm', () => {
  const props = {
    onloadAvatarAction: jest.fn(),
    prerenderAvatarURL: jest.fn(),
    handleSubmit: jest.fn(),
    isLoading: false,
  };

  const contactsForm = shallow(<ContactsForm {...props} />);

  it('snapshot', () => {
    expect(toJson(contactsForm)).toMatchSnapshot();
  });

  it('should catch submit', () => {
    contactsForm.find('form').simulate('submit');
    expect(props.handleSubmit.mock.calls.length).toBe(1);
  });
});
