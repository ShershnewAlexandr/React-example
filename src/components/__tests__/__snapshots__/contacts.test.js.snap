// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contactsForm snapshot 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    direction="row"
    spacing={4}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={6}
    >
      <Field
        component={[Function]}
        disabled={false}
        label="contacts.firstname"
        name="firstName"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={6}
    >
      <Field
        component={[Function]}
        disabled={false}
        label="contacts.lastname"
        name="lastName"
        type="date"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <Field
        component={[Function]}
        disabled={false}
        label="contacts.dateOfBirthday"
        name="dateOfBirthday"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <Field
        component={[Function]}
        disabled={false}
        label="contacts.photo"
        name="photo"
        onClickAction={[MockFunction]}
        prerenderPhotoURL={[MockFunction]}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</form>
`;
